Runtime error with 0.6.2 provider

After trying to upgrade from 0.6.1 to 0.6.2 we ran into the following error:

│ Error: Request cancelled
│ 
│   with data.opslevel_service.opslevel_services["<SERVICE_ID> (Removed for Github issue)"],
│   on main.tf line 5, in data "opslevel_service" "opslevel_services":
│    5: data "opslevel_service" "opslevel_services" {
│ 
│ The plugin.(*GRPCProvider).ValidateDataResourceConfig request was cancelled.
Releasing state lock. This may take a few moments...

Stack trace from the terraform-provider-opslevel_v0.6.2 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x19a95d8]

goroutine 147 [running]:
github.com/opslevel/terraform-provider-opslevel/opslevel.datasourceServiceRead(0x1c78bf6?, 0x18?)
	github.com/opslevel/terraform-provider-opslevel/opslevel/datasource_opslevel_service.go:141 +0x398
github.com/opslevel/terraform-provider-opslevel/opslevel.wrap.func1(0xc000272b70?, {0x1c60d40?, 0xc000ac7d60?})
	github.com/opslevel/terraform-provider-opslevel/opslevel/common.go:27 +0x36
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).ReadDataApply(0xc000886be0, 0xc000215110?, {0x1c60d40, 0xc000ac7d60})
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:413 +0x66
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).ReadDataApply(0xc000411880, 0xc000e6da60, 0x1b0bb60?)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:451 +0x65
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ReadDataSource(0xc00012d190, {0xc000437b40?, 0x10f2546?}, 0xc000437b40)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:1046 +0x2cc
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ReadDataSource_Handler({0x1c2c7e0?, 0xc00012d190}, {0x1f6c388, 0xc000a97ec0}, 0xc000c5cfc0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3341 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00034f880, {0x1f6fee8, 0xc000702d00}, 0xc000832480, 0xc00098a570, 0x264b9f0, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1283 +0xcfd
google.golang.org/grpc.(*Server).handleStream(0xc00034f880, {0x1f6fee8, 0xc000702d00}, 0xc000832480, 0x0)
	google.golang.org/grpc@v1.47.0/server.go:1620 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.47.0/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.47.0/server.go:920 +0x28a

Error: The terraform-provider-opslevel_v0.6.2 plugin crashed!

--- a/.changes/unreleased/Bugfix-20221215-055650.yaml
+++ b/.changes/unreleased/Bugfix-20221215-055650.yaml
@@ -1,0 +1,3 @@
+kind: Bugfix
+body: Fix nil pointer dereference on datasource opslevel_service
+time: 2022-12-15T05:56:50.575606-06:00

--- a/opslevel/datasource_opslevel_service.go
+++ b/opslevel/datasource_opslevel_service.go
@@ -138,7 +138,7 @@ func datasourceServiceRead(d *schema.ResourceData, client *opslevel.Client) erro
 	if err := d.Set("api_document_path", resource.ApiDocumentPath); err != nil {
 		return err
 	}
-	if err := d.Set("preferred_api_document_source", string(*resource.PreferredApiDocumentSource)); err != nil {
+	if err := d.Set("preferred_api_document_source", resource.PreferredApiDocumentSource); err != nil {
 		return err
 	}
 

